queenbee/io/reference.py

```diff
@@ -0,0 +1,315 @@
+"""Objects to reference parameters, files and folders from inputs, tasks and items."""
+
+import re
+from typing import List, Union, Dict
+from pydantic import Field, constr, validator
+
+from ..base.basemodel import BaseModel
+
+
+def template_string(keys: List[str]) -> str:
+    """Generate a template string from a list of key
+
+    Arguments:
+        keys {List[str]} -- A list of keys
+
+    Returns:
+        str -- A template string
+    """
+    return f'{{{{{".".join(keys)}}}}}'
+
+
+class _BaseReference(BaseModel):
+    """A Base reference model."""
+
+    type: constr(regex='^_BaseReference$') = '_BaseReference'
+
+    @property
+    def source(self):
+        return None
+
+    def fetch_from_flow(self, flow):
```

Comment:
Good question but I was thinking to ask that from you! 😁 It was in the original code and I just copy/pasted it. Since there was not enough information on what it does I didn't dare to delete it. Do you know if you use it in any shape or form?
If not we should just remove it.